src/rgw/rgw_admin.cc

```diff
@@ -4333,6 +4333,10 @@ int main(int argc, const char **argv)
   if (!user_id.empty() || !subuser.empty()) {
     ret = user.init(store, user_op);
     if (ret < 0) {
+      if (opt_cmd == OPT_USER_CREATE && user_op.is_preserved_user()) {
+          cerr << "uid: " << user_id << " is preserved by radosgw,"
+	       << " please choose another uid." << std::endl;
+      }
       cerr << "user.init failed: " << cpp_strerror(-ret) << std::endl;
       return -ret;
```

Comment:
@mikulely This is unrelated to your fix: Here `ret` is negative inside `if (ret < 0) {`. But is it making ret positive by putting a `-` sign ? Please ignore this comment if it is intentional.